src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -186,6 +207,93 @@ MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree,
 
 
 /*
+ * ReplaceExternParamsInOriginalQuery replaces the external parameters that appears
+ * in the query with the corresponding entries in the boundParams.
+ *
+ * Note that this function is inspired by eval_const_expr() on Postgres. We cannot
+ * use that function because it requires access to PlannerInfo.
+ */
+static Node *
+ReplaceExternParamsInOriginalQuery(Node *inputNode, ParamListInfo boundParams)
+{
+	int parameterIndex = 0;
+
+	Assert(boundParams != NULL);
+
+	if (inputNode == NULL)
+	{
+		return NULL;
+	}
+
+	if (IsA(inputNode, Param))
+	{
+		Param *paramToProcess = (Param *) inputNode;
+		ParamExternData correspondingParameterData;
+		int numberOfParameters = boundParams->numParams;
+		int parameterId = paramToProcess->paramid;
+		int16 typeLength = 0;
+		bool typeByValue = false;
+		Datum constValue = 0;
+		bool paramIsNull = false;
+
+		if (paramToProcess->paramkind != PARAM_EXTERN)
+		{
+			return inputNode;
+		}
+
+		if (parameterId < 0)
+		{
+			return inputNode;
+		}
```

Comment:
Optional: If you'd move the rest of the current scope into something like
```
if (paramToProcess->paramkind != PARAM_EXTERN && parameterId < 0)
{
}
```
you could move most  variable declarations inside, directly defining them, instead of using some pointless assignment first.